sentry_streams/src/gcs_writer.rs

```diff
@@ -55,35 +59,45 @@ impl GCSWriter {
 
         let client = client.default_headers(headers).build().unwrap();
 
-        GCSWriter { client, url }
+        GCSWriter { client, url, route }
     }
+}
 
-    pub fn write_to_gcs(
-        &self,
-        message: Message<RoutedValue>,
-    ) -> Result<Message<RoutedValue>, SubmitError<RoutedValue>> {
+impl TaskRunner<RoutedValue, RoutedValue, anyhow::Error> for GCSWriter {
+    fn get_task(&self, message: Message<RoutedValue>) -> RunTaskFunc<RoutedValue, anyhow::Error> {
         let client = self.client.clone();
         let url = self.url.clone();
-        let bytes = Python::with_gil(|py| pybytes_to_bytes(&message, py)).unwrap();
+        let route = message.payload().route.clone();
+        let actual_route = self.route.clone();
+
+        let bytes = traced_with_gil("writing to gcs", |py| pybytes_to_bytes(&message, py)).unwrap();
 
-        let res = client.post(&url).body(bytes).send();
+        Box::pin(async move {
+            if route != actual_route {
+                return Ok(message);
+            }
 
-        let response = res.unwrap();
-        let status = response.status();
+            let res: Result<reqwest::Response, reqwest::Error> =
+                client.post(&url).body(bytes).send().await;
```

Comment:
How was this working before without await ?

Reply:
Since we were using `RunTask` which requires a synchronous task to be run, I had to use the crate `blocking reqwest` and used its respective `Client`, `ClientBuilder` 

Now I am using the async `reqwest` crate